deepchecks/vision/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains the domain classifier drift check."""
+from collections import OrderedDict
+from typing import Any, List
+
+from deepchecks.core import CheckResult, DatasetKind
+from deepchecks.vision import Context, TrainTestCheck
+from deepchecks.core.check_utils.whole_dataset_drift_utils import run_whole_dataset_drift
+import pandas as pd
+
+__all__ = ['WholeDatasetDrift']
+
+DEFAULT_IMAGE_PROPERTIES = ['aspect_ratio',
+                            'area',
+                            'brightness',
+                            'normalized_red_mean',
+                            'normalized_green_mean',
+                            'normalized_blue_mean']
+
+
+class WholeDatasetDrift(TrainTestCheck):
+    """
+    Calculate drift between the entire train and test datasets using a model trained to distinguish between them.
+
+    Check fits a new model to distinguish between train and test datasets, called a Domain Classifier.
+    The Domain Classifier is a tabular model, that cannot run on the images themselves. Therefore, the check calculates
+    properties for each image (such as brightness, aspect ratio etc.) and uses them as input features to the Domain
+    Classifier.
+    Once the Domain Classifier is fitted the check calculates the feature importance for the domain classifier
+    model. The result of the check is based on the AUC of the domain classifier model, and the check displays
+    the change in distribution between train and test for the top features according to the
+    calculated feature importance.
+
+    Parameters
+    ----------
+    alternative_image_properties : List[str] , default: None
```

Comment:
No custom callables?

Reply:
Thought the user can insert them into their ImageFormatter class.

Reply:
Mmm doesn't seem comfortable to me - they need to subclass, when the only reason to subclass is to add a few methods? Passing callable seems more natural,

Reply:
I get it. 
But my reasons were:
1. If we want caching in the future, we need these functions as part of a class. And it makes sense that if a user passed alternative properties here, they'll do that in the other checks that use those as well.
2. In the case of passing callables, they need to pass both callable and str (func name). This makes for a really weird structure, of either receiving a list of str or a dict of str: callable. 

I agree my solution is not perfect, but not sure what I should do.